controllers/workflowdef/image.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2023 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workflowdef
+
+import (
+	"github.com/kiegroup/kogito-serverless-operator/api/metadata"
+	"github.com/kiegroup/kogito-serverless-operator/api/v1alpha08"
+	"github.com/kiegroup/kogito-serverless-operator/version"
+)
+
+const (
+	latestImageTag              = "latest"
+	nightlySuffix               = "nightly"
+	defaultWorkflowDevModeImage = "quay.io/kiegroup/kogito-swf-devmode"
+	defaultWorkflowBuilderImage = "quay.io/kiegroup/kogito-swf-builder"
+)
+
+// GetWorkflowAppImageNameTag retrieve the tag for the image based on the Workflow based annotation, <workflowid>:latest otherwise
+func GetWorkflowAppImageNameTag(w *v1alpha08.KogitoServerlessWorkflow) string {
+	v := w.Annotations[metadata.Version]
+	if v != "" {
+		return w.Name + ":" + v
+	}
+	return w.Name + ":" + latestImageTag
+}
+
+func GetDefaultWorkflowDevModeImageTag() string {
+	return getDefaultImageTag(defaultWorkflowDevModeImage)
+}
+
+func GetDefaultWorkflowBuilderImageTag() string {
+	return getDefaultImageTag(defaultWorkflowBuilderImage)
+}
+
+func getDefaultImageTag(imgTag string) string {
+	if version.IsSnapshot() {
```

Comment:
isn't better to use stringBuilder here?

Reply:
I don't think so. It's moved from another PR, I won't change it.